src/firewall/functions.py

```diff
@@ -188,6 +253,8 @@ def check(entry, *, family=None, flags=0, types):
         return True
 
 
+EntryTypeAddr = EntryType("addr", ipaddr_parse, ipaddr_unparse)
```

Comment:
Instead of passing parse/unsparse functions like this I'd rather see us derive the EntryType class and make these part of that class.

Reply:
`ipaddr_parse()` is a useful, stand-alone function that parses an IP address. If you know that you have a string and want to parse an IP address from it, then you just call `ipaddr_parse(str)`, and not bother with `EntryTypeAddr.parse(str)`.

The EntryType instances are useful, when we have a string which could be one of several things, like a port or an IP address. So you call `EntryType.parse(str, types=(EntryTypeAddr, EntryTypePort))`.  The result is the `(entry_type, detail)` which tells you the type it is (EntryTypeAddr or EntryTypePort) and the details of the parse operation (which is specific to the entry_type.

Anyway. So we need `ipaddr_parse` as stand alone function. At that point,
```
EntryTypeAddr = EntryType("addr", ipaddr_parse, ipaddr_unparse)
```
is the simplest (shortest) possible way to create those entry-types.

Also, subclassing is wrong EntryType is a parser/unparser/validator/normalizer for strings. I don't want subclasses of the parser (a ParserType for Addresses, a ParserType for Ports). We need *instances* of the parser.